### ISSUE TITLE
Filedownloader download give up retry exception 

### ISSUE THREAD

**Issue by Jeffin21** [lingochamp/FileDownloader#905]
filedownloader download give up retry exception 
I've an URL sometimes that server support multi threading and sometimes not so add a new function to reset the collection thread count to 1 when multi thread not support

**Comment by rantianhua** [comment]
Do you use FileDownloader#setMaxNetworkThreadCount method to reset the connect thread count?

**Comment by Jeffin21** [comment]
Yes, I used that and it makes all download tasks into single thread . I need to make single thread for some speficed task only

**Comment by rantianhua** [comment]
What's the returned value of that method, true or false? You need to note that this method will not work if there are tasks downloading.

**Comment by Jeffin21** [comment]
Actually the problem occurs when the download gets an error downloading with  multi-threading , please do a function to retry download with single threading before going to  download give up retry exception

**Comment by Jeffin21** [comment]
Is there anyway or you guys can add new function for this problem ? Some URLs will show multithread support before download start , when the download starts it won’t supporting multithread so please add a new function in download retry give up class to retry download with single thread before making an download error

**Comment by rantianhua** [comment]
Sorry for a very late response. Any new feature may not be added in FileDownloader, we are evaluating your request, please be patient.

**Comment by Jacksgong** [comment]
@Jeffin21

Thanks for your advice, I know what you want, but this feature does not make sense.

1. FileDownloader has already covered the case of some URL can't support multithread download. **Detail:**  before starting real downloading, we tried to connect to the backend with the URL and ask whether it can support `range` or whether `etag` is overdue, this is the main job of the `DownloadLaunchRunnable`.
2. If the backend of the URL you provided response it supports multithread downloading and `etag` is valid, but the downloading is failed just because of we using multithread, please ask your backend company to resolve the real issue.
3. If the URL backend still tell FileDownload please feel free to download it through multithread, but it always failed with multithread and you can't contact your company to resolve such problem, you also can force the URL downloading with single-thread through customize your own [`ConnectionCountAdapter`](https://github.com/lingochamp/FileDownloader/wiki/Customizable-Component) and maybe you need use `BaseDownloadTask#forceRedownload` to achieve that.
